### ISSUE TITLE
FIM issues in %WINDIR%\System32 directory

### ISSUE THREAD

**Issue by adi5353** [wazuh/wazuh#14796]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.3.7     |Manager&Agent |AllInOne| Packages | Windows 10/11|

Hello Team, We have observed that Syscheck FIM Module is not able to detect any changes in `%WINDIR%\System32` but it is able to show the changes in `%WINDIR%\Sysnative\Drivers\etc`.

The configuration used is the following:
```
<agent_config>
	<syscheck>
		<disabled>no</disabled>
		<scan_on_start>yes</scan_on_start>
		<!-- Frequency that syscheck is executed default every 12 hours -->
		<frequency>300</frequency>
		<!-- Default files to be monitored. -->
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="regedit.exe$|system.ini$|win.ini$">%WINDIR%</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\test\test.txt</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\test</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\Sysnative\Drivers\etc</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\SysNative</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\sysnative</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\System32</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\system32</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\System32\drivers\etc</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="WMIC.exe$">%WINDIR%\System32\wbem</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="powershell.exe$">%WINDIR%\System32\WindowsPowerShell\v1.0</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="winrm.vbs$">%WINDIR%\System32</directories>
		<!-- 32-bit programs. -->
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\System32\drivers\etc</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\system32</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\Sysnative</directories>
		<directories check_all="yes" whodata="yes" report_changes="yes" recursion_level="0">%WINDIR%\sysnative</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0">%WINDIR%\System32</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="WMIC.exe$">%WINDIR%\System32\wbem</directories>
		<directories check_all="yes" realtime="yes" report_changes="yes" recursion_level="0" restrict="powershell.exe$">%WINDIR%\System32\WindowsPowerShell\v1.0</directories>
		<ignore>%PROGRAMDATA%\Microsoft\Windows\Start Menu\Programs\Startup\desktop.ini</ignore>
		<registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services</registry_ignore>
		<registry_ignore type="sregex">\Enum$</registry_ignore>
		<ignore type="sregex">.log$|.htm$|.jpg$|.png$|.chm$|.pnf$|.evtx$|.dll$|.exe$|.dat$|.sam$</ignore>
		<!-- Frequency for ACL checking (seconds) -->
		<windows_audit_interval>60</windows_audit_interval>
		<!-- Nice value for Syscheck module -->
		<process_priority>10</process_priority>
		<!-- Maximum output throughput -->
		<max_eps>100</max_eps>
		<!-- Database synchronization settings -->
		<synchronization>
			<enabled>yes</enabled>
			<interval>5m</interval>
			<max_interval>1h</max_interval>
			<max_eps>10</max_eps>
		</synchronization>
	</syscheck>
</agent_config>
```

In the Inventory, We are able to see the System32 files but not the one which we have added for the testing. Kind of below SS:

![image](https://user-images.githubusercontent.com/100297998/188409638-d1c0a418-8994-4907-92d9-caf5d3b87c07.png)
![image](https://user-images.githubusercontent.com/100297998/188409669-0c264140-9ec7-434a-9ed5-6905cc3834d6.png)

With `report_changes` we can manage to get alerts of changes in the directories (otherwise we are not getting them), but everything gets complicated when we want to monitor `%WINDIR%\System32`, we have tried with `whodata` also, but the alerts are of policy failure and not of FIM, also does not reflect in the inventory.

Also tried it with the specific path `c:\windows\*` but with no success.

Let me know if you need more information.

Regards
Aditya Sharma

**Comment by GabrielEValenzuela** [comment]
# Solution report

## Goal
This report has for goal to make a summary of the possible solution.

## Issue research
This issue was solved 2 years ago. According to the solution posted in Slack, Mauro wrote: 
_With the `system32` directory is that it is quite particular, when coexisting 32 and 64 bits applications, Windows maintains 2 copies of these directories with 32 and 64 bits versions of their binaries. But there are some directories (like drivers) that for being of configuration only, exist in only one of the two and Windows makes changes underneath in the calls of the apps.
As the agent is 32 bits, the calls with `%WINDIR%%\System32` are translated to `%WINDIR%\Syswow64`, that is why it does not generate the alerts, it is scanning a different directory. To monitor `%WINDIR%\System32` in scheduled, you have to configure it as **`%WINDIR%\Sysnative`**_

Accessing the 64-bit **System32** folder from a 32-bit application by simply inserting `\System32` in the folder path is not possible. A redirection to the **SysWOW64** folder is made automatically by the system. So the solution is use `Sysnative`, a virtual folder, a special alias, that can be used to access the 64-bit **System32** folder from a 32-bit application or script. [^1]

## Conclutions
The proposed solution is to update the documentation perhaps using this [issue](https://github.com/wazuh/wazuh-documentation/issues/3586) to avoid the incorrect path.

[^1]: [The 'Sysnative' folder in 64-bit Windows explained](https://www.samlogic.net/articles/sysnative-folder-64-bit-windows.htm)

**Comment by adi5353** [comment]
Hi @GabrielEValenzuela,
I have tested it in the environment and it is working fine, but the new issue I have seen in the env. like below:

I created one file in the system32 folder and then the File was added to the system alert generated but after some time automatically on a scheduled basis, that file got deleted, so I am not sure if it's something to do with a windows machine or something else is happening.
Can you please give any ideas over this once?

![Screenshot_2](https://user-images.githubusercontent.com/100297998/194520650-d175f564-1d04-4ec9-b719-521604dad034.png)

Regards
Aditya Sharma

**Comment by GabrielEValenzuela** [comment]
Hi @adi5353 ! How are you ?

We have done some tests based on your previous comment.
We created a folder inside the `System32` path, and deleted all the **`ossec.conf`** entries, to set up a new
incrementally, adding new configuration modes in each test and observing if the file was deleted or not. 
the file was deleted or not.

I attach a screenshot with the six tests performed, which have the corresponding attributes of the FIM configuration used.

![image](https://user-images.githubusercontent.com/27935340/195026715-0636f493-07ad-49e2-983c-0605222bbb54.png)


In conclusion, our team decides in a first approximation that this could be a problem related to 
with the configuration used in the **`ossec.conf`**, but we are still looking into it.

Thanks for your time and have a nice day 🙂 

Best,

**Comment by adi5353** [comment]
Hi Team,

I have tested it and it is working fine. I am closing this issue for now.

Regards
Aditya Sharma

**Comment by chemamartinez** [comment]
Hi @adi5353,

I will reopen this issue since we are working on a fix to make this behavior more consistent.

**Comment by adi5353** [comment]
Hi @chemamartinez , I am checking it in the environment once again and found the same issue of automatic deletion of the file.

![image](https://user-images.githubusercontent.com/100297998/198183854-6e2285ce-ff60-4916-af94-a18d52a7d99f.png)

```
<agent_config>
<syscheck>
<disabled>no</disabled>
<scan_on_start>yes</scan_on_start>
<!-- Frequency that syscheck is executed default every 12 hours -->
<frequency>60</frequency>
<!-- Default files to be monitored. -->
<directories check_all="yes" realtime="yes" whodata="yes" recursion_level="0" restrict="regedit.exe$|system.ini$|win.ini$">%WINDIR%</directories>
<directories check_all="yes" realtime="yes" whodata="yes" recursion_level="0">%WINDIR%\sysnative\Drivers\etc</directories>
<directories check_all="yes" realtime="yes" whodata="yes" recursion_level="0">%WINDIR%\sysnative</directories>
<directories check_all="yes" realtime="yes" recursion_level="0" restrict="WMIC.exe$">%WINDIR%\System32\wbem</directories>
<directories check_all="yes" realtime="yes" recursion_level="0" restrict="powershell.exe$">%WINDIR%\System32\WindowsPowerShell\v1.0</directories>
<directories check_all="yes" realtime="yes" recursion_level="0" restrict="winrm.vbs$">%WINDIR%\System32</directories>
<!-- 32-bit programs. -->
<directories check_all="yes" realtime="yes" whodata="yes" recursion_level="0">%WINDIR%\sysnative</directories>
<directories check_all="yes" realtime="yes" whodata="yes" recursion_level="0">%WINDIR%</directories>
<directories check_all="yes" realtime="yes" recursion_level="0" restrict="WMIC.exe$">%WINDIR%\System32\wbem</directories>
<directories check_all="yes" realtime="yes" recursion_level="0" restrict="powershell.exe$">%WINDIR%\System32\WindowsPowerShell\v1.0</directories>
<ignore>%PROGRAMDATA%\Microsoft\Windows\Start Menu\Programs\Startup\desktop.ini</ignore>
<ignore>%WINDIR%\Sysnative\Drivers\etc\hosts</ignore>
<registry_ignore>HKEY_LOCAL_MACHINE</registry_ignore>
<registry_ignore type="sregex">\Enum$</registry_ignore>
<ignore type="sregex">.log$|.htm$|.jpg$|.png$|.chm$|.pnf$|.evtx$|.dll$|.dat$|.sam$|.exe$</ignore>
<!-- Frequency for ACL checking (seconds) -->
<windows_audit_interval>60</windows_audit_interval>
<!-- Nice value for Syscheck module -->
<process_priority>10</process_priority>
<!-- Maximum output throughput -->
<max_eps>100</max_eps>
<!-- Database synchronization settings -->
<synchronization>
<enabled>yes</enabled>
<interval>5m</interval>
<max_interval>1h</max_interval>
<max_eps>10</max_eps>
</synchronization>
</syscheck>
</agent_config>
```

**Comment by T100D** [comment]
Question about this, does this mean that when a file like c:\windows\sysnative\drivers\etc\hosts is changed this is visible in the inventory page, but not registered as an event, because this is what I see after setting up Wazuh (4.3.9) and installing the agent on a windows 10 environment (Microsoft Windows 10 Enterprise 10.0.19044) ?

**Comment by adi5353** [comment]
Hi @GabrielEValenzuela ,

Is there any other solution instead of these syntax changes and all?

Regards
Aditya Sharma
